master/internal/rm/kubernetesrm/kubernetes_resource_manager_intg_test.go

```diff
@@ -0,0 +1,242 @@
+//go:build integration
+// +build integration
+
+package kubernetesrm
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	k8sV1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/determined-ai/determined/master/internal/config"
+	"github.com/determined-ai/determined/master/internal/mocks"
+	"github.com/determined-ai/determined/master/pkg/device"
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+const (
+	defaultNamespace    = "default"
+	auxNodeName         = "aux"
+	compNodeName        = "comp"
+	pod1NumSlots        = 4
+	pod2NumSlots        = 8
+	nodeNumSlots        = 8
+	nodeNumSlotsCPU     = 20
+	podServSlotTypeGPU  = "randomDefault"
+	cpuResourceRequests = int64(4000)
+)
+
+// Number of active slots on given nodes.
+var nodeToSlots = map[string]int{
+	auxNodeName:  pod1NumSlots,
+	compNodeName: pod2NumSlots,
+}
+
+func TestGetAgentsAndGetSlots(t *testing.T) {
+	// Create resource list.
+	resourceList := map[k8sV1.ResourceName]resource.Quantity{
+		k8sV1.ResourceName(ResourceTypeNvidia): *resource.NewQuantity(
+			int64(nodeNumSlots),
+			resource.Format("DecimalSI")),
+		k8sV1.ResourceCPU: *resource.NewQuantity(
+			int64(nodeNumSlotsCPU),
+			resource.Format("DecimalSI")),
+	}
+
+	// Create auxiliary agent node and compute node.
+	auxNode := k8sV1.Node{
+		ObjectMeta: metaV1.ObjectMeta{
+			ResourceVersion: "1",
+			Name:            auxNodeName,
+		},
+		Status: k8sV1.NodeStatus{Allocatable: resourceList},
```

Comment:
why do the aux and compute nodes both look like they have GPUs? i know this is a test, but it is strange.